### PR TITLE
Update GH workflows to test PHP 7.3 and above and bump oldest compatibility test to WC 7.7.0

### PR DESCRIPTION
**Pull request by mattallan**
## Description

Our GH checks have been failing with the following error:
image

tl;dr During our PHP 7.1 test, we're trying to install the latest Gutenberg which requires PHP 7.2 and that's causing the test to fail.

Since Woo Subscriptions supports WooCommerce 7.7+ which requires PHP 7.3, this PR updates our workflows to:
- remove the PHP 7.1, 7.2, 7.3 when testing the latest version of WC
- updates our "Oldest supported version" to test against WooCommerce 7.7 with PHP 7.3

## How to test this PR

1. Confirm the GH actions pass.

## Product impact

- [ ] Added changelog entry (or does not apply)
- [ ] Will this PR affect WooCommerce Subscriptions? yes/no/tbc, add issue ref
- [ ] Will this PR affect WooCommerce Payments? yes/no/tbc, add issue ref
- [ ]  Added deprecated functions, hooks or classes to the [spreadsheet](https://docs.google.com/spreadsheets/d/1xw9xszcPMnWsp4C8OKZMsLzZob7tOmWT7qMqmEIq314/edit#gid=0)

### CODE REVIEW

**Comment by mattallan** [comment]
GH checks are passing now. Merging this 😎
